packages/core/src/external/carequality/ihe-gateway-v2/inbound/__tests__/process-dq-request.test.ts

```diff
@@ -0,0 +1,117 @@
+import { metriportOrganization } from "@metriport/shared";
+import { InboundDocumentQueryResp } from "@metriport/ihe-gateway-sdk";
+import { createITI38SoapEnvelope } from "../../outbound/xca/create/iti38-envelope";
+import { processDqResponse } from "../../outbound/xca/process/dq-response";
+import { processInboundDqRequest } from "../xca/process/dq-request";
+import { createInboundDqResponse } from "../xca/create/dq-response";
+import {
+  TEST_CERT,
+  TEST_KEY,
+  xcaGateway,
+  outboundDqRequest,
+} from "../../outbound/__tests__/constants";
+import { signTimestamp } from "../../saml/security/sign";
+import { createExtrinsicObjectXml } from "../../../dq/create-metadata-xml";
+import { extractDocumentUniqueId } from "../../../shared";
+
+describe("Process Inbound Dq Request", () => {
+  it("should process successful Iti-38 request", () => {
+    try {
+      const soapEnvelope = createITI38SoapEnvelope({
+        bodyData: outboundDqRequest,
+        publicCert: TEST_CERT,
+      });
+      const signedEnvelope = signTimestamp({ xml: soapEnvelope, privateKey: TEST_KEY });
+      const iti38Request = processInboundDqRequest(signedEnvelope);
+      expect(iti38Request.externalGatewayPatient).toEqual(outboundDqRequest.externalGatewayPatient);
+    } catch (error) {
+      throw new Error("iti38Request externalPatient is wrong or undefined");
+    }
+  });
+
+  it("should process invalid ITI-38 request correctly", () => {
+    const soapEnvelope = createITI38SoapEnvelope({
+      bodyData: outboundDqRequest,
+      publicCert: TEST_CERT,
+    });
+    expect(() => {
+      processInboundDqRequest(soapEnvelope);
+    }).toThrow("Failed to parse ITI-38 request");
+  });
+});
+
+describe("Process Inbound Dq Response", () => {
+  it("should process successful Iti-38 response", () => {
+    const docUniqueId = "1234567890";
+    const extrinsicObjectXmls = [
+      createExtrinsicObjectXml({
+        createdTime: new Date().toISOString(),
+        organization: metriportOrganization,
+        size: "1000",
+        patientId: "1234567890",
+        documentUniqueId: "1234567890",
+        mimeType: "application/xml",
+      }),
+    ];
+    const response: InboundDocumentQueryResp = {
+      ...outboundDqRequest,
+      responseTimestamp: new Date().toISOString(),
+      extrinsicObjectXmls,
+    };
+
+    const xmlResponse = createInboundDqResponse(response);
+    const iti38Response = processDqResponse({
+      response: {
+        gateway: xcaGateway,
+        outboundRequest: outboundDqRequest,
+        success: true,
+        response: xmlResponse,
+      },
+    });
+    if (iti38Response.documentReference && iti38Response.documentReference[0]) {
+      expect(extractDocumentUniqueId(iti38Response.documentReference[0].docUniqueId)).toEqual(
+        docUniqueId
+      );
+    } else {
+      throw new Error(
+        "iti38Response.documentReference is undefined or has wrong document unique id"
+      );
+    }
```

Comment:
Not a huge deal, but I'd keep the main flow/indentation as part of the test, and have alternative flows on conditionals, like

```
... // prepare/execute
if (unexpected condition) error;
... // check results